kerastuner/tuners/bayesian.py

```diff
@@ -0,0 +1,285 @@
+import numpy as np
+import random
+import json
+from scipy import optimize as scipy_optimize
+from sklearn import gaussian_process
+
+from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
+from ..engine import tuner as tuner_module
+from ..engine import oracle as oracle_module
+from ..engine import hyperparameters as hp_module
+
+
+class BayesianOptimizationOracle(oracle_module.Oracle):
+    """Bayesian optimization oracle.
+
+    It uses Bayesian optimization with a underlying Gaussian process model.
+    The acquisition function used is upper confidence bound (UCB).
+
+    Attributes:
+        init_samples: Int. The number of randomly generated samples as initial
+            training data for Bayesian optimization.
+        alpha: Float. Value added to the diagonal of the kernel matrix
+            during fitting.
+        beta: Float. The balancing factor of exploration and exploitation.
+            The larger it is, the more explorative it is.
+        seed: Int. Random seed.
+
+    """
+
+    def __init__(self,
+                 init_samples=2,
```

Comment:
It is not immediately clear that `init` should be a number, or that "samples" refers to sets of hp values in the search space. What could be a better name? `num_initial_configurations`? `num_initial_points`?

Reply:
I think `num_initial_points` is better.